int32 count_goal
---
bool success
---
int32 current_number